cli/src/rpc.rs

```diff
@@ -152,62 +180,71 @@ impl<S: Serialization, C: Send + Sync + 'static> RpcBuilder<S, C> {
 	}
 
 	/// Builds into a usable, sync rpc dispatcher.
-	pub fn build(self) -> RpcDispatcher<S, C> {
+	pub fn build(self, log: log::Logger) -> RpcDispatcher<S, C> {
 		RpcDispatcher {
-			i: Arc::new(self),
-			calls: Arc::new(std::sync::Mutex::new(HashMap::new())),
+			log,
+			context: self.context,
+			calls: self.calls,
+			serializer: self.serializer,
+			methods: Arc::new(self.methods),
 		}
 	}
 }
 
 type DispatchMethod = Box<dyn Send + Sync + FnOnce(Outcome)>;
 
 /// Dispatcher returned from a Builder that provides a transport-agnostic way to
-/// deserialize and handle RPC calls. This structure may get more advanced as
+/// deserialize and dispatch RPC calls. This structure may get more advanced as
 /// time goes on...
-pub struct RpcDispatcher<S, C> {
-	i: Arc<RpcBuilder<S, C>>,
+#[derive(Clone)]
+pub struct RpcCaller<S: Serialization> {
+	serializer: Arc<S>,
 	calls: Arc<Mutex<HashMap<u32, DispatchMethod>>>,
+	sender: mpsc::UnboundedSender<Vec<u8>>,
 }
 
-impl<S, C> Clone for RpcDispatcher<S, C> {
-	fn clone(&self) -> Self {
-		RpcDispatcher {
-			i: self.i.clone(),
-			calls: self.calls.clone(),
-		}
-	}
-}
+impl<S: Serialization> RpcCaller<S> {
```

Comment:
Likewise, the RPCDispatcher was split into a caller as well. The dispatcher is owned by the 'read loop' on the socket, but the callers can be passed and cloned around to other consumers.